Automattic/vip-go-mu-plugins
# Prevent WP_User_Query multi dataset queries on sites with multiple datasets

## Description

In order to support multiple datasets, the following args are not allowed on `WP_User_Query`:
* `has_published_posts`
* `orderby = post_count`

This PR introduces a filter that intercepts the `WP_User_Query` and in case one of the above args is present, it filters it out and adds a warning log.

This filter is only enabled for sites with multiple datasets enabled.

## Pre-review checklist

Please make sure the items below have been covered before requesting a review:


## Pre-deploy checklist


## Steps to Test
1. Check out PR.
2. Define the following constant:
```php
define( 'VIP_IS_MULTIPLE_DATASETS_ENABLED', true );
```
3. Run the following query:
```php
$users_query = new \WP_User_Query(
	array(
		'has_published_posts' => true,
		'orderby'             => 'post_count',
	)
);
$users_query->get_results();

```
